lib/gui/app.js

```diff
@@ -96,7 +96,7 @@ app.run((AnalyticsService, UpdateNotifierService, SelectionStateModel) => {
       if (!isLatestVersion && !SelectionStateModel.hasImage()) {
 
         AnalyticsService.logEvent('Notifying update');
-        UpdateNotifierService.notify();
+        UpdateNotifierService.notify().catch(ErrorService.reportException);
```

Comment:
Can we return this promise and put the `.catch` block in `UpdateNotifierService.isLatestVersion()`? `UpdateNotifierService.isLatestVersion()` might throw something in the future, and we will be already prepared for it!

```js
UpdateNotifierService.isLatestVersion().then((isLatestVersion) => {

  // In case the internet connection is not good and checking the
  // latest published version takes too long, only show notify
  // the user about the new version if he didn't start the flash
  // process (e.g: selected an image), otherwise such interruption
  // might be annoying.
  if (!isLatestVersion && !SelectionStateModel.hasImage()) {

    AnalyticsService.logEvent('Notifying update');
    return UpdateNotifierService.notify();
  }
}).catch(ErrorService.reportException);
```